src/Arcus.Scripting.Management/Scripts/Remove-AzApiManagementSoftDeletedService.ps1

```diff
@@ -0,0 +1,38 @@
+Param(
+    [Parameter(Mandatory = $true)][string] $Name = $(throw "Name of the API Management instance is required"),
+    [Parameter(Mandatory = $false)][string] $SubscriptionId = "",
+    [Parameter(Mandatory = $false)][string] $AccessToken = ""
+)
+
+if($SubscriptionId -eq "" -or $AccessToken -eq ""){
+    # Request accessToken in case the script contains records
+    $token = Get-AzCachedAccessToken
+
+    $AccessToken = $token.AccessToken
+    $SubscriptionId = $token.SubscriptionId
+}
+
+$authHeader = @{
+   'Authorization'='Bearer ' + $AccessToken
+}
+
+Write-Host "Checking if the API Management instance with name '$Name' is listed as a soft deleted service"
+$getUri = 'https://management.azure.com/subscriptions/{0}/providers/Microsoft.ApiManagement/deletedservices?api-version=2021-08-01' -f $SubscriptionId
```

Comment:
We should probably also make this URL configurable based on the cloud. (Like, global cloud, Germany...). I think we already have something like this in Logic Apps, so you can check. Can be done in a separate PR, though.